packages/ComponentLibrary/src/components/Drawer/DrawerSection/index.tsx

```diff
@@ -1,94 +1,82 @@
-import React, { useState } from 'react';
+import { useCallback, useMemo, useState } from 'react';
 import { Collapse, Box, Typography } from '@mui/material';
 import { ExpandLess, ExpandMore } from '@mui/icons-material';
 import { styles } from '../styles';
 import { theme } from '../../../theme';
-import { Section } from '../types';
+import DrawerSubsection from '../DrawerSubsection';
+import { DrawerSectionProps } from '../types';
 
-const DrawerSection: React.FC<any> = ({
-  section,
-  open,
-  level = 0,
-  onSelect,
-}) => {
-  // Constants
-  const isMainSection = level === 0;
-  const isSelected =
-    section.isSelected ||
-    section.subSections?.some((subsection: Section) => subsection.isSelected);
-
-  // States
+const DrawerSection = ({ item, onClick }: DrawerSectionProps<Menu>) => {
+  const isMainSection = item.type === 'folder';
+  const isSelected = false;
   const [expanded, setExpanded] = useState(false);
 
-  // Handlers
-  const handleClick = () => {
-    if (section.subSections) {
+  const handleClick = useCallback(() => {
+    if (item.type === 'folder') {
       setExpanded(prev => !prev);
-    }
-
-    if (section.isSelected || (section.subSections && expanded)) {
-      onSelect(null);
     } else {
-      onSelect(section.id);
+      console.warn('DrawerSection: unexpected type');
     }
-  };
+  }, [item]);
 
-  return (
-    <Box
-      sx={{
-        ...styles.drawerSectionBox,
-        backgroundColor:
-          isMainSection && isSelected
-            ? theme.palette.baselineColor.neutral[10]
-            : 'transparent',
-        padding:
-          isMainSection && !!section.subSections && isSelected && '0.5rem',
-      }}>
-      <Box
-        onClick={handleClick}
-        sx={{
-          ...styles.listItemButton,
-          ...(section.isSelected && styles.listItemButtonSelected),
-          ...styles.listItemContentText,
-          borderRadius: open ? '0.5rem' : '12.5rem',
-          fontSize: open ? '1.5rem' : '1rem',
-          justifyContent: open ? 'space-between' : 'center',
-        }}>
-        <Box
-          sx={{
-            ...styles.listItemInnerContentText,
-            justifyContent: open ? 'flex-start' : 'center',
-          }}>
-          <Box sx={styles.listItemIconContent}>
-            <Typography sx={styles.listItemIconTypography}>
-              {section.icon}
-            </Typography>
-          </Box>
+  const mainStyle = useMemo(
+    () => ({
+      ...styles.drawerSectionBox,
+      backgroundColor:
+        isMainSection && isSelected
+          ? theme.palette.baselineColor.neutral[10]
+          : 'transparent',
+      padding:
+        isMainSection && !!item.submenu && isSelected ? '0.5rem' : undefined,
+    }),
+    [isMainSection, isSelected, item.submenu],
```

Comment:
Is it necessary to useMemo to change styles?

Is the calculation required too heavy?